dataactbroker/helpers/generation_helper.py

```diff
@@ -398,101 +352,72 @@ def check_generation_prereqs(submission_id, file_type):
-        # Check to see if the same file exists in the child bucket
-        s3 = boto3.client('s3', region_name=CONFIG_BROKER["aws_region"])
-        response = s3.list_objects_v2(Bucket=CONFIG_BROKER['aws_bucket'], Prefix=child_job.filename)
-        for obj in response.get('Contents', []):
-            if obj['Key'] == child_job.filename:
-                # The file already exists in this location
-                log_data['message'] = 'Cached {} file CSV already exists in this location'.format(file_type)
-                logger.info(log_data)
-                return
-
-        # Copy the parent file into the child's S3 location
-        log_data['message'] = 'Copying the cached {} file from job {}'.format(file_type, parent_job.job_id)
-        logger.info(log_data)
-        S3Handler.copy_file(CONFIG_BROKER['aws_bucket'], CONFIG_BROKER['aws_bucket'], parent_job.filename,
-                            child_job.filename)
-
-
-def update_validation_job_info(sess, job):
-    """ Populates validation job objects with start and end dates, filenames, and status.
-        Assumes the upload Job's start and end dates have been validated.
-    """
-    if job.submission_id:
-        # Retrieve and update the validation Job
+    # Change the validation job's file data when within a submission
+    if job.submission_id is not None:
         val_job = sess.query(Job).filter(Job.submission_id == job.submission_id,
                                          Job.file_type_id == job.file_type_id,
                                          Job.job_type_id == lookups.JOB_TYPE_DICT['csv_record_validation']).one()
-        val_job.start_date = job.start_date
-        val_job.end_date = job.end_date
-        val_job.filename = job.filename
-        val_job.original_filename = job.original_filename
-        val_job.job_status_id = lookups.JOB_STATUS_DICT["waiting"]
-
-        # Clear out error messages to prevent stale messages
-        val_job.error_message = None
-
-    job.error_message = None
+        val_job.filename = filename
+        val_job.original_filename = original_filename
+
+        # Copy the data to the Submission's bucket
+        if not is_local and file_generation.file_path != job.filename:
+            # Check to see if the same file exists in the child bucket
+            s3 = boto3.client('s3', region_name=CONFIG_BROKER["aws_region"])
+            bucket = CONFIG_BROKER['aws_bucket']
+            response = s3.list_objects_v2(Bucket=bucket, Prefix=job.filename)
+            for obj in response.get('Contents', []):
+                if obj['Key'] == job.filename:
+                    # The file already exists in this location
+                    log_data['message'] = '{} file already exists in this location: {}'.format(job.file_type.name,
+                                                                                               job.filename)
+                    logger.info(log_data)
+                    return
```

Comment:
So if there should be no collision, why have the case logged where the file already exists? 

I would think we want 1 of 2 results in this scenario:
1) Throw an error because this scenario should never exist
2) Replace the existing file with the one just generated, because it should be more up-to-date (and log that this is being done)

Reply:
1. Generate a new D file (detached or within-submission).
2. Create a Submission and generate a matching D file so it uses the cached version from step 1.
3. Regenerate D files within the same Submission.

Step 2 will pull the cached file from step 1 into the Submission's bucket. Step 3 will try to do that once more, but see that the file already exists (with the same timestamp) and therefore not re-copy the file over. The only time that we'd attempt to place the same file in the bucket more than once in this situation.

Reply:
Ok, I see. So it's informationally logging the fact that no work needs to be done, since we already stored the necessary file.

I did not originally understand that from the log statement of just
> `{} file already exists in this location: {}`

That statement makes me think something is wrong. Could you adjust it slightly to be more informative, like:

_Found previously-generated file. Able to skip copy of same (cached) file to this location, since the original one can be reused._

or something like that.